pandas/tests/arithmetic/test_timedelta64.py

```diff
@@ -1236,6 +1236,66 @@ def test_td64arr_rfloordiv_tdlike_scalar(self, scalar_td, box_with_array):
         res = tdi // (scalar_td)
         tm.assert_equal(res, expected)
 
+    # ------------------------------------------------------------------
+    # mod, divmod
+    # TODO: operations with timedelta-like arrays, numeric arrays,
+    #  reversed ops
+
+    def test_td64arr_mod_tdscalar(self, box_with_array, three_days):
+        tdi = timedelta_range('1 Day', '9 days')
+        tdarr = tm.box_expected(tdi, box_with_array)
+
+        expected = TimedeltaIndex(['1 Day', '2 Days', '0 Days'] * 3)
+        expected = tm.box_expected(expected, box_with_array)
+
+        result = tdarr % three_days
+        tm.assert_equal(result, expected)
+
+        if box_with_array is pd.DataFrame:
+            pytest.xfail("DataFrame does not have __divmod__ or __rdivmod__")
```

Comment:
You have two distinct tests here as delineated by this `xfail` check.  In the interest of modularity, this test should be broken up into IMO.  Similar comment goes for your other tests where there is an `xfail` like this.